### DISCUSSION TITLE
Dynamically position dropdowns to avoid viewport edge

### DISCUSSION THREAD

**Discussion by ghost** [tailwindlabs/headlessui#1210] [Ideas]
I believe it would be a valuable usability improvement to reposition the dropdown when it approaches the viewport edge. My primary use case occurs when you use any kind of dropdown near the bottom of a scrollable form. The dropdown extends the page rather than popping above the trigger element, and it is an annoyance to use because the page jerks when you open/close the dropdown and/or requires scrolling to view the dropdown contents.

Bootstrap dropdowns are dynamic:
https://getbootstrap.com/docs/5.1/components/dropdowns/#single-button

**Comment by RobinMalfait** [comment]
Hey! Thank you for your suggestion!
Much appreciated! 🙏

You can use a tool like https://floating-ui.com for this in combination with Headless UI 👍
